### PR TITLE
Ventus/refactor

### LINKED ISSUE #26: Build backend function for getting session summaries

### PR DESCRIPTION
**Pull request by KenjiPcx**
### Summary of changes
- Move db repository code in backend to its own layer for better architecture
- Refactor backend functions code style
- Flesh out analyze session api

### CODE REVIEW

File: frontend/src/components/ArticleCard.module.css
Comment (KenjiPcx) [inline frontend/src/components/ArticleCard.module.css:1]: @VentusTWY we switched to mantine v6, so module css not working anymore 😢, but we can use chatgpt to convert to v6 syntax, ask it to generate the css classes object using the `createStyles` api

**Review by KenjiPcx — COMMENTED** [review]
@VentusTWY You can review the code to see my changes?

**Comment by KenjiPcx** [comment]
Closes #26

**Comment by KenjiPcx** [comment]
I didn't touch on the frontend though, wasn't sure if you work on it on a separate branch

**Comment by VentusTWY** [comment]
merged for now , commented out prompt template imports on backend

image

image

**Pull request merged**